casepro/msgs/tasks.py

```diff
@@ -89,3 +103,117 @@ def reply_export(export_id):
     logger.info("Starting replies export #%d..." % export_id)
 
     ReplyExport.objects.get(pk=export_id).do_export()
+
+
+def create_faq(org, question, answer, language, parent, labels=(), **kwargs):
+    """
+    A helper for creating FAQs since labels (many-to-many) needs to be added after initial creation
+    """
+    faq = FAQ.objects.create(org=org, question=question, answer=answer, language=language, parent=parent, **kwargs)
+    faq.labels.add(*labels)
+    return faq
+
+
+def get_or_create_lang(org, lang_code):
+    """
+    Gets or creates a Language object from a potentially poorly formatted language code
+    """
+    lang_code = lang_code.strip()
+    try:
+        # Get the Language
+        return Language.objects.get(code__iexact=lang_code)  # iexact removes case sensitivity
+    except:
+        # Format the lang_code before creating a new language
+        if len(lang_code) != 6:
+            raise ValueError("Language codes should be 6 characters long")
+
+        if lang_code[3] != '_':
+            raise ValueError("Language name and location should be seperated by an underscore")
+
+        lang_code = lang_code[0:4] + lang_code[4:6].upper()
+        # Create the language
+        return Language.objects.create(org=org, code=lang_code)
+
+
+def get_labels(task, labelstring):
+    """
+    Gets a list of label objects from a comma-seperated string of the label codes, eg. "TB, aids"
+    """
+    labels = set()
+    labelstrings = labelstring.split(', ')
+    for labelstring in labelstrings:
+        labelstring = labelstring.strip()
+
+        try:
+            label = Label.objects.get(name__iexact=labelstring)  # iexact removes case sensitivity
+            labels.add(label)
+        except Exception as e:
+            task.log("Label %s does not exist" % labelstring)
+            raise e
+    return list(labels)
+
+
+@task(track_started=True)
+def faq_csv_import(org, task_id):  # pragma: no cover
```

Comment:
Is there a reason we can't use smartmin's CSV import stuff? Shouldn't require much more than making `FAQ` implement `SmartModel` and implementing `Smartmodel.create_instance`